python-package/lightgbm/dask.py

```diff
@@ -331,6 +486,7 @@ def _train(
             worker_address_to_port=worker_address_to_port,
             time_out=params.get('time_out', 120),
             return_model=(worker == master_worker),
+            evals_provided=eval_set is not None,
```

Comment:
instead of passing through a flag, could this just be computed inside `_train`? I'd like to limit the use of passed-through flags to control internal behavior, if possible. I've found in the past that that creates unnecessary risk of mistakes.

Reply:
Yeah so this flag is trying to communicate to each worker whether or not an `eval_set` was provided to the dask estimator, so that if any worker was not provided any eval_data, it should fail out (each worker needs at least a little bit). So I can add this flag as an element of each `part` instead, but just an FYI it's not as clean as a single flag one-liner.

Reply:
OH I see, ok I misunderstood. I think it's ok to leave here